### PR TITLE
[YUNIKORN-1687] add e2e for user & group quota enforcement

### PR DESCRIPTION
**Pull request by FrankYang0529**
### What is this PR for?

- [X] Single specific user limit
  - [X] A user can't create more than maxapplications.
  - [X] A user can't create more than maxresoruces.
  - [X] A user can create apps in other queues.
  - [X] Users not in the limit can create more than maxapplications.
  - [X] Users not in the limit can create more than maxresources.
- [X] Single specific group limit
  - [X] A group can't create more than maxapplications.
  - [X] A group can't create more than maxresoruces.
  - [X] A group can create apps in other queues.
  - [X] Groups not in the limit can create more than maxapplications.
  - [X] Groups not in the limit can create more than maxresources.
- [X] User limit is lower than group limit.
- [X] Group limit is lower than user limit.

### What type of PR is it?
* [X] - Improvement

### What is the Jira issue?

https://issues.apache.org/jira/browse/YUNIKORN-1687

### How should this be tested?

All e2e test cases pass.

### CODE REVIEW

File: test/e2e/user_group_limit/user_group_limit_test.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+		Ω(sandboxResourceUsageDAO.ResourceUsage).NotTo(gomega.BeNil())
+		Ω(sandboxResourceUsageDAO.ResourceUsage.Resources["pods"]).To(gomega.Equal(resources.Quantity(1)))
+		Ω(len(sandboxResourceUsageDAO.RunningApplications)).To(gomega.Equal(1))
+		Ω(sandboxResourceUsageDAO.RunningApplications[0]).To(gomega.Equal(sleepPod1Application))
+		Ω(sandboxResourceUsageDAO.RunningApplications[0]).NotTo(gomega.Equal(sleepPod2Application))
+
+		// // sleepPod2 can't be scheduled because application count is more than maxapplications
+		// ginkgo.By("The sleep pod " + sleepPod2.Name + " can't be scheduled")
+		// err = kClient.WaitForPodUnschedulable(sleepPod2, 60*time.Second)
+		// gomega.Ω(err).NotTo(gomega.HaveOccurred())
```
Comment (FrankYang0529) [inline test/e2e/user_group_limit/user_group_limit_test.go:148]: Not sure whether I understand User Based Quota Enforcement correctly. If the `maxapplications` is 1, I expect a user can't create more than one application in a queue. However, the second application can be scheduled, but the related resource usage is not tracked in the user tracker. From the source code, it looks like we don't handle the result from [IncreaseTrackedResource](https://github.com/apache/yunikorn-core/blob/0ffd1392c329b185bdd8e51665d48e0502807892/pkg/scheduler/objects/application.go#L1656).

@wilfred-s, may you give me some feedback about this? Thank you!

**Comment by manirajv06** [comment]
@FrankYang0529 How should we take this up? Should we cover all cases described in https://issues.apache.org/jira/browse/YUNIKORN-1871 and plan for review? I think better to cover all the cases and review together.

**Comment by FrankYang0529** [comment]
> @FrankYang0529 How should we take this up? Should we cover all cases described in https://issues.apache.org/jira/browse/YUNIKORN-1871 and plan for review? I think better to cover all the cases and review together.

Hi @manirajv06, yes, I agree it's better to cover all the cases in this PR. I will update the PR and remove WIP once it's ready for review. I will do my best to finish it this week. Thank you!

**Comment by FrankYang0529** [comment]
> > @FrankYang0529 How should we take this up? Should we cover all cases described in https://issues.apache.org/jira/browse/YUNIKORN-1871 and plan for review? I think better to cover all the cases and review together.
> 
> Hi @manirajv06, yes, I agree it's better to cover all the cases in this PR. I will update the PR and remove WIP once it's ready for review. I will do my best to finish it this week. Thank you!

@manirajv06 , I have to create another PR (https://github.com/apache/yunikorn-core/pull/595) for YUNIKORN-1871, because it's unit test in yunikorn-core. I will finish the unit test first and then back to this PR for e2e test. Thank you.

**Comment by manirajv06** [comment]
@FrankYang0529 Now that https://github.com/apache/yunikorn-core/pull/595 has gone in, can you clean up this pr? and also, can you ensure all cases has been covered as e2e tests?

**Comment by FrankYang0529** [comment]
> @FrankYang0529 Now that [apache/yunikorn-core#595](https://github.com/apache/yunikorn-core/pull/595) has gone in, can you clean up this pr? and also, can you ensure all cases has been covered as e2e tests?

Hi @manirajv06, sorry for the late reply. I will update test cases in this weekend. Thank you.

**Comment by FrankYang0529** [comment]
Hi @manirajv06, I would like to split wildcard and config change cases to another two Jira, because I don't make this PR too big. Thank you.

**Review by manirajv06 — APPROVED** [review]

**Comment by FrankYang0529** [comment]
Created another two Jira. Thank you.
* [YUNIKORN-1956](https://issues.apache.org/jira/browse/YUNIKORN-1956): Add wildcard user/group limit e2e tests
* [YUNIKORN-1957](https://issues.apache.org/jira/browse/YUNIKORN-1957): Add user/group limit config change e2e test

**Pull request closed without merging**